source/includes/connection-snippets/x509-snippets/csharp-connection.cs

```diff
@@ -0,0 +1,48 @@
+// begin x509 connection
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Net.Security;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.Core;
+using System;
+using System.Threading.Tasks;
+
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using System.Security.Permissions;
+
+namespace WorkingWithMongoDB
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            MainAsync().Wait();
+            Console.WriteLine("done");
+        }
+
+        static async Task MainAsync()
+        {
+            var settings = new MongoClientSettings 
+            {
+                Credential =  MongoCredential.CreateMongoX509Credential(null),
+                SslSettings = new SslSettings
+                {
+                    ClientCertificates = new List<X509Certificate>()
+                    {
+                        new X509Certificate2("/etc/certs/mongodb/client-certificate.pfx", "<pfx_passphrase>")
+                    },
+                },
+                UseTls = true,
+                Server = new MongoServerAddress("<cluster-url>")
+            };
+
+            var client = new MongoClient(settings);
+        }
```

Comment:
It might be better to use the sample code from the recently revised C# documentation for X509. It's not yet live, but you can see the changes made in github.

In particular, let's show starting from a connection string and adding in via code the things that can only be added via code:

https://github.com/mongodb/mongo-csharp-driver/commit/ee9f21e423ffb02e8c42c5c7192b352403800499#diff-c2ca42c0112df581679f2ede5989ee0aR83

Reply:
Sure we've now updated the code to be very similar to that link. Let me know what you think